### DISCUSSION TITLE
rtl-sdr 4 on Pi Zero?

### DISCUSSION THREAD

**Discussion by SFSailor** [jvde-github/AIS-catcher#343] [Q&A]
Thanks for your ongoing support. [My project](https://pcup.org/aisreport.php) is going well, beta testing with RTL-SDR 4 on a Pi 5 and RTL-SDR 3 on a Pi Zero.

For cost and power reasons, I'd like to build the next few nodes on a Pi Zero, but the prebuilt code won't run the RTL-SDR 4, and the code won't compile on the Pi Zero (it hangs during the build)

Is there a secret to compiling on the Zero, or can the prebuilt be tweaked to support the RTLSDR 4?

For those interested, project page is here: https://pcup.org/aisreport.php

**Comment by jvde-github** [comment]
Is this the Zero or the Zero 2W? I am asking for the chipset, the RPI1 has limited floating support. 

One way is perhaps to cross-compile in a Docker and statically link the RTL-SDR library in the executable. Is not an easy one.

**Comment by SFSailor** [comment]
Sorry, it's the Zero 2 W

**Reply by jvde-github** [reply]
So, in principle the Debian package works and runs. Issue is that SDR V4 is not supported. Perhaps I can include it in the distribution without too much hassle.

**Comment by SFSailor** [comment]
The debian package works and runs great!  If it can include the SDR V4 that would be great.

As long as we are at it, is there any way to include microseconds in the received time? As long as I am getting the same AIS report at multiple locations, I may as well take a whack at TDOA

**Comment by jvde-github** [comment]
Ok, the Debian packages should now support the RTL-SDR V4. I have statically linked the library in the executable and build that from source. I can however not test if it now works with the V4. So, feedback appreciated.

Microseconds is possible. Will put it on the to-do list.

**Reply by SFSailor** [reply]
Oh awesome!  I'll burn a fresh OS and give it a try!

**Reply by SFSailor** [reply]
Hmm... does it still work with v3? I just overwrote the old code with the new and it's not seeing the existing V3 plugged in...

**Reply by SFSailor** [reply]
Works great with the V4!

**Reply by jvde-github** [reply]
Great. What about your V3 comment? Assume that has been resolved.

Thanks for checking!

**Reply by SFSailor** [reply]
I'll swap back in the morning to see if V3 runs now. Have you made any updates to the code since the intial V4?

**Reply by jvde-github** [reply]
Nope. Code is the same, only thing that changed is the latest version of the RTL-SDR library from Osmocom included in the package. I tested on my site with the V3 and it worked normally. Just could not test with a V4 on a running station.

**Reply by SFSailor** [reply]
Okay, after a few reboots it's working fine with the V3 as well!  Good work.

**Comment by tkreyche** [comment]
The Pi 4B/2GB works great and isn't that much more than the Zero 2W, especially when you consider the overall cost of dongle, cases, antenna, cables. I admire Jasper's programming skill to get it to run on such a constricted device but wonder if it's a good expenditure of time/effort. Anyway I have a Zero 2W lying around someplace and will give it a try.

**Reply by SFSailor** [reply]
Thanks.  I have a pi 4 on santa's list.  That's a likely direction if the Z2W is not an easy fix.
